#Estudio de caso 2
- Manuel Antonio Cartín Hernández
##Carga del data set
- Importa el archivo en un DataFrame de pandas y muestra las primeras 5 filas.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
from scipy.stats import shapiro
# Leer el archivo
df = pd.read_csv('sample_data/vgsales.csv')
print(df.head())

   Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0       NaN     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  


#2. Selección estratégica de datos
-	Extrae solo las columnas Rank, Name, Platform, Genre, NA_Sales y Global_Sales, ya que serán clave para nuestro análisis.

In [16]:
# Seleccionar las columnas de interés
selected_columns = df[['Rank', 'Name', 'Platform', 'Genre','NA_Sales', 'Global_Sales']]
print(selected_columns.head())

# Extraer la columna 'release_year' y convertirla a un array de NumPy
Global_Sales = df['Global_Sales'].values
Global_Sales_array = np.array(Global_Sales)
print(Global_Sales_array)


   Rank                      Name Platform         Genre  NA_Sales  \
0     1                Wii Sports      Wii        Sports       NaN   
1     2         Super Mario Bros.      NES      Platform     29.08   
2     3            Mario Kart Wii      Wii        Racing     15.85   
3     4         Wii Sports Resort      Wii        Sports     15.75   
4     5  Pokemon Red/Pokemon Blue       GB  Role-Playing     11.27   

   Global_Sales  
0         82.74  
1         40.24  
2         35.82  
3         33.00  
4         31.37  
[8.274e+01 4.024e+01 3.582e+01 ... 1.000e-02 1.000e-02 1.000e-02]


#Limpieza de datos
-	Elimina las filas con valores nulos para garantizar la calidad del análisis. (Si no la identificas puedes preguntarle al profesor)

In [11]:
# Verificar valores faltantes
missing_values = selected_columns.isnull().sum()
print(missing_values)


Rank            0
Name            0
Platform        0
Genre           0
NA_Sales        4
Global_Sales    0
dtype: int64


In [12]:
#eliminar valor nulo
# Eliminar filas con valores faltantes en la columna 'Year'
cleaned_data = selected_columns.dropna(subset=['NA_Sales'])
print(cleaned_data)


        Rank                                              Name Platform  \
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
6          7                             New Super Mario Bros.       DS   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

              Genre  NA_Sales  Global_Sales  
1          Platform     29.08         40.24  
2      

In [13]:
# Verificar valores faltantes
missing_values_cleaned = cleaned_data.isnull().sum()
print(missing_values_cleaned)


Rank            0
Name            0
Platform        0
Genre           0
NA_Sales        0
Global_Sales    0
dtype: int64


#Transformacion de datos
-	Asegúrate de que la columna Global_Sales tenga el tipo de dato adecuado para facilitar cálculos numéricos. (float)

In [17]:
# Convertir la columna 'Year' a enteros
cleaned_data['Global_Sales'] = cleaned_data['Global_Sales'].astype(float)
print(cleaned_data)
Global_Sales_float= Global_Sales_array.astype(float)
print(Global_Sales_float)


        Rank                                              Name Platform  \
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
6          7                             New Super Mario Bros.       DS   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

              Genre  NA_Sales  Global_Sales  
1          Platform     29.08         40.24  
2      

#Filtrado Inteligente
-	Selecciona todos los videojuegos lanzados exclusivamente en la plataforma PS3.


In [15]:
# Filtrar los datos para extraer juegos publicados en la plataforma 'Wii'
PS3_Games = cleaned_data[cleaned_data['Platform'] == 'PS3']
print(PS3_Games.head())


    Rank                            Name Platform    Genre  NA_Sales  \
16    17              Grand Theft Auto V      PS3   Action      7.01   
34    35      Call of Duty: Black Ops II      PS3  Shooter      4.99   
37    38  Call of Duty: Modern Warfare 3      PS3  Shooter      5.54   
40    41         Call of Duty: Black Ops      PS3  Shooter      5.98   
54    55                  Gran Turismo 5      PS3   Racing      2.96   

    Global_Sales  
16         21.40  
34         14.03  
37         13.46  
40         12.73  
54         10.77  


#Calculo metricas Clave
-	Calcula el promedio de ventas globales para todos los videojuegos y analiza qué tan alto o bajo es este valor.

In [29]:
# Calcular la media de la columna 'Global_Sales'
global_sales_mean = PS3_Games['Global_Sales'].mean()
print(f"La media de Global_Sales es: {global_sales_mean}")


La media de Global_Sales es: 0.7207223476297968


#Hallazgo del más vendido
- Identifica el videojuego con mayores ventas globales y conviértelo en el rey de la lista.

In [25]:
# Encuentra el índice del videojuego con más ventas globales
indice_rey = PS3_Games['Global_Sales'].idxmax()
videojuego_rey = PS3_Games.loc[indice_rey]
print("El videojuego con más ventas globales es:\n", videojuego_rey)



El videojuego con más ventas globales es:
 Rank                            17
Name            Grand Theft Auto V
Platform                       PS3
Genre                       Action
NA_Sales                      7.01
Global_Sales                  21.4
Name: 16, dtype: object


# Hallazgo del Menos Vendido:
-	Encuentra el videojuego con menos ventas globales y analiza posibles razones detrás de este resultado.


In [26]:
# Encuentra el índice del videojuego con menos ventas globales
indice_menos_vendido = PS3_Games['Global_Sales'].idxmin()
videojuego_menos_vendido = PS3_Games.loc[indice_menos_vendido]
print("El videojuego con menos ventas globales es:\n", videojuego_menos_vendido)


El videojuego con menos ventas globales es:
 Rank                                              16017
Name            Nobunaga no Yabou Online: Houou no Shou
Platform                                            PS3
Genre                                          Strategy
NA_Sales                                            0.0
Global_Sales                                       0.01
Name: 16014, dtype: object


# Análisis por Género:
- Agrupa los datos por género de videojuego y calcula la suma total de ventas para cada categoría.


In [27]:
# Agrupar por 'Genre' y calcular la suma de ventas
ventas_por_genero = df.groupby('Genre')[['NA_Sales', 'Global_Sales']].sum()
print(ventas_por_genero)


              NA_Sales  Global_Sales
Genre                               
Action          877.83       1751.18
Adventure       105.80        239.04
Fighting        223.59        448.91
Misc            396.21        809.96
Platform        437.51        831.37
Puzzle          100.58        244.95
Racing          359.42        732.04
Role-Playing    327.28        927.37
Shooter         582.60       1037.37
Simulation      183.31        392.20
Sports          641.86       1330.93
Strategy         68.70        175.12


# Búsqueda de un Clásico:
- Encuentra todos los videojuegos cuyo nombre contenga la palabra "Mario" y descubre qué tan influyente ha sido este personaje en la industria.


In [28]:
# Filtrar los datos para extraer juegos publicados en la plataforma 'Wii'
Mario_Busqueda = PS3_Games[PS3_Games['Name'] == 'Mario']
print(Mario_Busqueda.head())


Empty DataFrame
Columns: [Rank, Name, Platform, Genre, NA_Sales, Global_Sales]
Index: []


#Analisis
en el PS3_games referente a los juegos vendidos para la consola de sony se dan las siguientes estadisticas:
- una media de global sales es de 0.72
- el más vendido es grand thef auto V
- el menos vendido es nobunaga no yabou online: houou no shou.
- 877.83 y 1751. 18 son la suma total de ventas en genero accion
- 105.80 y 239.04 para el genero aventura
- 223.59 y 448.91 para genero pelea
- 396.21 y 809.96 para el genero misc
- 437.51 y 831.37 para el genero platform
- 100.58 y 244.95 para genero puzle
- 359.42 y 732.04 para el genro racing
- 327.28 y 927.37 para el genero role play
- 582.60 y 1037.37 para el genero disparo
- 183.31 y 392.20 para el genero simulador
- 641.86 y 1330.93 para el genero deporte
- 68.70  y 175.12 para el genero estrategia
todos los valores anterior mente definidos en los generos son primero las ventas NA y despues las ventas Globales
para ventas en el area NA son de genero accion las mas vendidas, junto con las de Global.
por ultimo no se encontro ningun referente de videojuego en el que se incluya Mario como personaje incluido en el titulo de un videojuego en la plataforma de sony.
